lib/newspaper_works/data/work_derivative_loader.rb

```diff
@@ -3,12 +3,27 @@
 module NewspaperWorks
   module Data
     class WorkDerivativeLoader
+      include Enumerable
+
       def initialize(work)
+        # context usually work, may be FileSet, may be string id of FileSet
         @context = work
         # storage for computed paths are memoized as used, here:
         @paths = {}
       end
 
+      # all paths for work derivatives
+      def paths
+        path_factory.derivatives_for_reference(fileset_id)
+      end
+
+      # enumerates available file extensions
+      def each
```

Comment:
I would maybe suggest a different custom method name here, rather than overriding `each`?

Reply:
Using Enumerable when possible seems AFAICT idiomatic for many Ruby classes (make things act enumerable when the they represent something enumerable or list-like)?  Is there a reason to avoid using Enumerable?

Reply:
It's typical for methods to return an enumerable object (e.g. `Array`), but overriding methods from the `Enumerable` class is less common. It seems slightly extraneous. What type of object does `paths` return?